### ISSUE TITLE
Kedro forces pip-tools~=6.5 and fails with new pip==23.2.0 & pip-tools==7.0.0

### ISSUE THREAD

**Issue by FloridaDataGuy** [kedro-org/kedro#2807]
## Description
Kedro forces pip-tools~=6.5 and fails with new pip==23.2.0 & pip-tools==7.0.0.
As I undestand pip-tools are fixed in dependencies/requriements.txt in Kedro package source.

## Context
Trying to use the latest combination of `pip==23.2.0` and `pip-tools==7.0.0`. Can not instantiate virtual environment if there is a requrement `kedro==0.18.11`. 

## Steps to Reproduce
1. enter into requrements.in pip-tools==7.0.0, kedro==0.18.11

2. install pip==23.2.0

3. run pip-compile

## Expected Result
Receive compiled requirements.txt.

## Actual Result
pip-compile fails with the following error:

```
ERROR: Cannot install kedro[pandas]==0.18.11 and pip-tools==7.0.0 because these package versions have conflicting dependencies.
```

## Your Environment

* Kedro version used (`pip show kedro` or `kedro -V`): 0.18.11
* Python version used (`python -V`): 3.10.12
* Operating system and version: Debian 11 (Bullseye)

**Comment by astrojuanlu** [comment]
Thanks for reporting @FloridaDataGuy , we'll have a look at this soon.

**Comment by noklam** [comment]
@astrojuanlu Anything `pip-tools` actually break in 7.0.0? Python 3.7 support is dropped from 7.0.0, is it possible to keep test running < 7.0.0 but allow people install with newer version?

**Comment by astrojuanlu** [comment]
IIRC this was fixed in #2843, see

https://github.com/kedro-org/kedro/blob/6913acdfd55898f956b6d91fc4602fbdb011a5d1/pyproject.toml#L31

**Comment by noklam** [comment]
Ah thanks, I thought we only update test requirements. I will close this issue then.
